src/Core/Content/Media/Aggregate/MediaThumbnail/MediaThumbnailEntity.php

```diff
@@ -68,6 +83,19 @@ public function setUrl(string $url): void
 
     public function getMediaId(): string
     {
+        if (!Feature::isActive('v6.8.0.0')) {
+            Feature::triggerDeprecationOrThrow('v6.8.0.0', '$mediaId must not be null');
+
+            if (!isset($this->mediaId)) {
+                return '';
+            }
+
+            return $this->mediaId;
+        }
+
+        /** @deprecated tag:v6.8.0 - Will be non-nullable */
+        \assert(\is_string($this->mediaId));
+
         return $this->mediaId;
```

Comment:
```suggestion
        if (!isset($this->mediaId)) {
            Feature::triggerDeprecationOrThrow('v6.8.0.0', '$mediaId must not be null');

            return '';
        }

        return $this->mediaId;
```

this should already be enough :thinking: if you keep it like this, you would not get an exception if `mediaId` is still `null` with 6.8

Reply:
Yes. `$mediaId` must **never** be null. It's allowed to be null for now because in 6.6 one service created thumbnails without a mediaId (which is also not intended in 6.6) and these got serialized in the database. We don't want thumbnails without a `$mediaId`, but need to allow null at least when deserialized or populated by `assign()` to not have to throw away 6.6 carts when migrating to 6.7.

And isn't it easier to wrap the deprecated code in a Feature flag so when 6.8 is prepares, the whole block can be removed?

Reply:
with your could it could be possible, that if the feature flag 6.8 is active, the mediaId could be `null` and then will get a type error, because the getter does not allow to return `null` 

instead the `triggerDeprecationOrThrow` should throw an exception stating that the `mediaId` cannot longer be `null` 

yes, normally we should wrap deprecated code with a feature flag to make it easier to remove, but imo in this case this could lead to wrong behaviour :thinking: